core/src/main/java/org/elasticsearch/common/geo/builders/ShapeBuilder.java

```diff
@@ -415,293 +386,57 @@ public static Orientation fromString(String orientation) {
-             */
-            if (coordinates.children.size() < 2) {
-                throw new ElasticsearchParseException("invalid number of points in LineString (found [{}] - must be >= 2)",
-                        coordinates.children.size());
-            }
-
-            CoordinatesBuilder line = new CoordinatesBuilder();
-            for (CoordinateNode node : coordinates.children) {
-                line.coordinate(node.coordinate);
-            }
-            return ShapeBuilders.newLineString(line);
-        }
-
-        protected static MultiLineStringBuilder parseMultiLine(CoordinateNode coordinates) {
-            MultiLineStringBuilder multiline = ShapeBuilders.newMultiLinestring();
-            for (CoordinateNode node : coordinates.children) {
-                multiline.linestring(parseLineString(node));
-            }
-            return multiline;
-        }
-
-        protected static LineStringBuilder parseLinearRing(CoordinateNode coordinates, boolean coerce) {
-            /**
-             * Per GeoJSON spec (http://geojson.org/geojson-spec.html#linestring)
-             * A LinearRing is closed LineString with 4 or more positions. The first and last positions
-             * are equivalent (they represent equivalent points). Though a LinearRing is not explicitly
-             * represented as a GeoJSON geometry type, it is referred to in the Polygon geometry type definition.
-             */
-            if (coordinates.children == null) {
-                String error = "Invalid LinearRing found.";
-                error += (coordinates.coordinate == null) ?
-                        " No coordinate array provided" : " Found a single coordinate when expecting a coordinate array";
-                throw new ElasticsearchParseException(error);
-            }
-
-            int numValidPts = coerce ? 3 : 4;
-            if (coordinates.children.size() < numValidPts) {
-                throw new ElasticsearchParseException("invalid number of points in LinearRing (found [{}] - must be >= [{}])",
-                        coordinates.children.size(), numValidPts);
-            }
-
-            if (!coordinates.children.get(0).coordinate.equals(
-                    coordinates.children.get(coordinates.children.size() - 1).coordinate)) {
-                if (coerce) {
-                    coordinates.children.add(coordinates.children.get(0));
-                } else {
-                    throw new ElasticsearchParseException("invalid LinearRing found (coordinates are not closed)");
-                }
-            }
-            return parseLineString(coordinates);
-        }
+    /** @deprecated this method will be removed in a future version; use ShapeParser.parse instead */
+    @Deprecated
+    public static ShapeBuilder parse(XContentParser parser) throws IOException {
+        DEPRECATION_LOGGER.deprecated("deprecated method [ShapeBuilder.parse] used; replaced by [ShapeParser.parse]");
```

Comment:
I don't think we need deprecation logging here, I think marking the java API should be enough.